evalml/pipelines/components/transformers/preprocessing/time_series.py

```diff
@@ -0,0 +1,42 @@
+import pandas as pd
+
+from evalml.pipelines.components.transformers.transformer import Transformer
+
+
+class DelayedFeaturesTransformer(Transformer):
+    name = "Delayed Features Transformer"
+    hyperparameter_ranges = {}
+    needs_fitting = False
+
+    def __init__(self, max_delay=2, random_state=0, **kwargs):
+
+        parameters = {"max_delay": max_delay}
+        parameters.update(kwargs)
+        super().__init__(parameters=parameters, random_state=random_state)
+        self.max_delay = max_delay
+
+    def fit(self, X, y=None):
+        """Fits the LaggedFeatureExtractor."""
+
+    def transform(self, X, y=None):
+        """Transforms the input data."""
+        if not isinstance(X, pd.DataFrame):
+            if y is None:
+                X = pd.DataFrame(X, columns=["target"])
+            else:
+                X = pd.DataFrame(X)
+        if y is not None and not isinstance(y, pd.Series):
+            y = pd.Series(y)
+
+        original_columns = X.columns
+        X = X.assign(**{f"{col}_delay_{t}": X[col].shift(t)
```

Comment:
If the time-series is irregularly spaced, shifting may give you data that is too "far in the past", e.g. we have daily data, but for some reason, a whole month is missing. We've decided that irregularly-spaced time series are out-of-scope for this first release. 

https://alteryx.quip.com/AM04ASOaQS4v/Time-Series-November-Design-Document